include/mxnet/c_lapack_api.h

```diff
@@ -54,6 +57,40 @@ extern "C" {
 
 inline char loup(char uplo, bool invert) { return invert ? (uplo == 'U' ? 'L' : 'U') : uplo; }
 
+
+/*!
+ * \brief Transpose matrix data in memory
+ * 
+ * Equivalently we can see it as flipping the layout of the matrix
+ * between row-major and column-major.
+ *
+ * \param m number of rows of input matrix a
+ * \param n number of columns of input matrix a
+ * \param b output matrix
+ * \param ldb leading dimension of b
+ * \param a input matrix
+ * \param lda leading dimension of a
+ */
+template <typename xpu, typename DType>
+inline void flip(int m, int n, DType *b, int ldb, DType *a, int lda);
```

Comment:
The `flip()` does "flipping" of elements in the memory. A row-major matrix will be "transposed" if we consider the result matrix still following the row-major layout; or alternatively, the result matrix will be "the same" as the input one if we consider the result matrix following column-major layout. Similar goes for a column-major input matrix.

Therefore there's no flag for the layout in this function's signature. All it does is mechanical "flipping" of elements. It's left to the developer how to interpret and use the result.